src/main/java/net/fabricmc/loom/configuration/providers/mappings/tiny/MappingsMerger.java

```diff
@@ -110,32 +109,11 @@ public static void legacyMergedMergeAndSaveMappings(Path from, Path out, Interme
 	 * Currently, Yarn does not export mappings for these inner classes.
 	 */
 	private static void inheritMappedNamesOfEnclosingClasses(MemoryMappingTree tree) {
-		int intermediaryIdx = tree.getNamespaceId("intermediary");
-		int namedIdx = tree.getNamespaceId("named");
+		assert tree.getNamespaceId("intermediary") > MappingTree.SRC_NAMESPACE_ID;
 
-		// The tree does not have an index by intermediary names by default
+		// Create an index by intermediary names for faster lookups during the propagation
 		tree.setIndexByDstNames(true);
 
-		for (MappingTree.ClassMapping classEntry : tree.getClasses()) {
-			String intermediaryName = classEntry.getDstName(intermediaryIdx);
-			String namedName = classEntry.getDstName(namedIdx);
-
-			if (intermediaryName.equals(namedName) && intermediaryName.contains("$")) {
-				String[] path = intermediaryName.split(Pattern.quote("$"));
-				int parts = path.length;
-
-				for (int i = parts - 2; i >= 0; i--) {
-					String currentPath = String.join("$", Arrays.copyOfRange(path, 0, i + 1));
-					String namedParentClass = tree.mapClassName(currentPath, intermediaryIdx, namedIdx);
-
-					if (!namedParentClass.equals(currentPath)) {
-						classEntry.setDstName(namedParentClass
-										+ "$" + String.join("$", Arrays.copyOfRange(path, i + 1, path.length)),
-								namedIdx);
-						break;
-					}
-				}
-			}
-		}
+		tree.propagateOuterClassNames("intermediary", List.of("named"), false);
```

Comment:
Passing `false` here for `processRemappedDstNames` matches the old code's behavior, but `true` would probably be preferable. Not sure if that actually changes anything for Yarn though.

Reply:
Lets keep it as-is, unless there is a good reason to change it.